src/utils/transaction.ts

```diff
@@ -314,6 +586,203 @@ export const buildCrossChainAction = async (
     }
   }
 
+  if (
+    transactionBlock.type === TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE &&
+    !!transactionBlock?.values?.fromChainId &&
+    !!transactionBlock?.values?.fromAsset &&
+    !!transactionBlock?.values?.amount &&
+    !!transactionBlock?.values?.offer &&
+    !!transactionBlock?.values?.receiverAddress &&
+    !!transactionBlock?.values?.toAsset
+  ) {
+    if (!transactionBlock?.values?.hasEnoughPLR) {
+      // Swap
+      try {
+        const {
+          values: {
+            amount,
+            fromChainId: chainId,
+            fromAsset: {
+              address: fromAssetAddress,
+              symbol: fromAssetSymbol,
+              decimals: fromAssetDecimals,
+              logoURI: fromAssetIconUrl,
+            },
+            toAsset: {
+              address: toAssetAddress,
+              symbol: toAssetSymbol,
+              decimals: toAssetDecimals,
+              logoURI: toAssetIconUrl,
+            },
+            offer,
+            receiverAddress,
+            accountType,
+          },
+        } = transactionBlock;
+
+        const fromAmountBN = ethers.utils.parseUnits(amount, fromAssetDecimals);
+
+        const swapServiceDetails = swapServiceIdToDetails[offer.provider];
```

Comment:
sorry I don't get it, is it always offer included in plr staking?

Reply:
Yes offer is required and we are handling offers in the main screen. We need offer object while doing transaction
![image](https://user-images.githubusercontent.com/109526304/216045340-fced9900-443e-42bd-a49d-39b030be786c.png)